File: packages/@xrnjs/app-utils/harmony/xrn_app_utils/src/main/ets/XRNFileModule.ts
import { TurboModule } from '@rnoh/react-native-openharmony/ts';
import { TM } from "@rnoh/react-native-openharmony/generated/ts"


export class XRNFileModule extends TurboModule implements TM.XRNFileModule.Spec {
  clearFrescoCache(): Promise<boolean> {
    //鸿蒙平台无需实现
    return new Promise((resolve, reject) => {
      resolve(true)
    })
  }

  insertImageToPhotoLibrary(path: string, fileName: string | null): Promise<boolean> {
    //鸿蒙平台无需实现
    return new Promise((resolve, reject) => {
      resolve(true)
    })
  }

}

File: packages/@xrnjs/app-utils/ios/XRNFileModule.swift
//
//  XRNFileModule.swift
//  xtapp
//
//  Created by  xtgq on 2025/5/8.
//  Copyright © 2025 Facebook. All rights reserved.
//

import Foundation

@objc(XRNFileModule)
class XRNFileModule: NSObject {

    @objc(clearFrescoCache:withRejecter:)
    func clearFrescoCache(resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        resolve(NSNumber(value: false))
    }
	
	@objc(insertImageToPhotoLibrary:fileName:withResolver:withRejecter:)
	  func insertImageToPhotoLibrary(_ path: String, fileName: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
		  resolve(NSNumber(value: false))
	}
  
}
